tests/Unit/Resources/ShippingResourceTest.php

```diff
@@ -0,0 +1,318 @@
+<?php
+
+namespace Jasara\AmznSPA\Tests\Unit\Resources;
+
+use Illuminate\Http\Client\Request;
+use Illuminate\Support\Str;
+use Jasara\AmznSPA\AmznSPA;
+use Jasara\AmznSPA\DataTransferObjects\Requests\Shipping\CreateShipmentRequest;
+use Jasara\AmznSPA\DataTransferObjects\Requests\Shipping\GetRatesRequest;
+use Jasara\AmznSPA\DataTransferObjects\Requests\Shipping\PurchaseLabelsRequest;
+use Jasara\AmznSPA\DataTransferObjects\Requests\Shipping\PurchaseShipmentRequest;
+use Jasara\AmznSPA\DataTransferObjects\Requests\Shipping\RetrieveShippingLabelRequest;
+use Jasara\AmznSPA\DataTransferObjects\Responses\Shipping\CancelShipmentResponse;
+use Jasara\AmznSPA\DataTransferObjects\Responses\Shipping\CreateShipmentResponse;
+use Jasara\AmznSPA\DataTransferObjects\Responses\Shipping\GetAccountResponse;
+use Jasara\AmznSPA\DataTransferObjects\Responses\Shipping\GetRatesResponse;
+use Jasara\AmznSPA\DataTransferObjects\Responses\Shipping\GetShipmentResponse;
+use Jasara\AmznSPA\DataTransferObjects\Responses\Shipping\GetTrackingInformationResponse;
+use Jasara\AmznSPA\DataTransferObjects\Responses\Shipping\PurchaseLabelsResponse;
+use Jasara\AmznSPA\DataTransferObjects\Responses\Shipping\PurchaseShipmentResponse;
+use Jasara\AmznSPA\DataTransferObjects\Responses\Shipping\RetrieveShippingLabelResponse;
+use Jasara\AmznSPA\Tests\Unit\UnitTestCase;
+
+class ShippingResourceTest extends UnitTestCase
+{
+    public function testCreateShipment()
+    {
+        list($config, $http) = $this->setupConfigWithFakeHttp('shipping/create-shipment');
+
+        $request = new CreateShipmentRequest(
+            client_reference_id: Str::random(),
+            ship_to: $this->setupShippingAddress(),
+            ship_from: $this->setupShippingAddress(),
+            containers: [
```

Comment:
I would move this into a setup method like the shipping address, too much data in the actual test method.